src/categoricalvector.jl

```diff
@@ -0,0 +1,85 @@
+
+export CategoricalVector
```

Comment:
Hmm, IIUC its only purpose is to treat it as a categorical axis, isn't it? Ideas about other possible names? It would be too bad to have conflicts when loading both AxisArrays and DataTables/DataFrames.